rasa_sdk/forms.py

```diff
@@ -689,3 +689,76 @@ def _get_entity_type_of_slot_to_fill(
                 return None
 
         return entity_type
+
+
+class FormValidationAction(Action):
+    """An action that validates if every extracted slot is valid."""
+
+    def name(self) -> Text:
+        """Unique identifier of this action.
+
+        Returns:
+            Name of the action.
+        """
+
+        raise NotImplementedError("An action must implement a name")
+
+    async def validate(
+        self,
+        dispatcher: "CollectingDispatcher",
+        tracker: "Tracker",
+        domain: "DomainDict",
+    ) -> List[EventType]:
+        """Validate slots by calling a validation function for each slot.
+
+        Args:
+            dispatcher: the dispatcher which is used to
+                send messages back to the user.
+            tracker: the state tracker for the current user.
+            domain: the bot's domain.
+        Returns:
+            A dictionary of `rasa_sdk.events.Event` instances.
+        """
+        slots: Dict[Text, Any] = tracker.form_slots_to_validate()
+
+        for slot_name, slot_value in slots.items():
+            function_name = f"validate_{slot_name}"
+            validate_func = getattr(self, function_name, lambda *x: False)
```

Comment:
optional: I think checking for a validation function as early as possible is a little bit clearer
```suggestion
            validate_func = getattr(self, function_name, None)
            if not validate_func:
                warnings.warn(...)
                continue
```